src/google/protobuf/compiler/java/java_file.cc

```diff
@@ -248,9 +248,12 @@ void FileGenerator::Generate(io::Printer* printer) {
   PrintGeneratedAnnotation(
       printer, '$', options_.annotate_code ? classname_ + ".java.pb.meta" : "");
   printer->Print(
-      "public final class $classname$ {\n"
+      "$deprecation$public final class $classname$ {\n"
```

Comment:
Is there a test to check this annotation?

Reply:
I only added a proto to make sure it compiles. Manually checked the annotations were generated though.

I can add a test to verify those annotations in a later PR (when I get time...)